src/components/TableDialog.vue

```diff
@@ -157,6 +187,88 @@ export default Vue.extend({
         this.fileName = this.fileName || getFileName(file);
         this.fileUploadError = null;
       }
+
+      interface TypeScore {
+        strings: Set<string>;
+        number: number;
+        date: number;
+        total: number;
+      }
+      const columnTypes = new Map<string, TypeScore>();
+
+      Papa.parse(file, {
+        header: true,
+        skipEmptyLines: true,
+        step(row: ParseResult<{}>) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const data = row.data as { [key: string]: any };
+
+          Object.keys(data).forEach((key: string) => {
+            if (!columnTypes.has(key)) {
+              columnTypes.set(key, {
+                strings: new Set<string>(),
+                number: 0,
+                date: 0,
+                total: 0,
+              });
+            }
+
+            const entry = columnTypes.get(key);
+            if (entry === undefined) {
+              throw new Error('impossible');
+            }
+
+            // Pass the value through a gauntlet of heuristics to see what types
+            // it can convert to.
+            const value = data[key];
+            entry.total += 1;
+
+            // Test for categoriness by counting up all the unique strings in
+            // the column.
+            entry.strings.add(value);
+
+            // See if the value can be converted to a number. Empty strings
+            // convert to 0, so those are excluded specifically.
+            if (value !== '' && !Number.isNaN(Number(value))) {
+              entry.number += 1;
+            }
+
+            // See if the value looks like a date.
+            if (dayjs(value).isValid()) {
+              entry.date += 1;
+            }
+          });
+        },
+
+        complete: () => {
+          const typeRecs = new Map<string, MultinetType>();
+
+          columnTypes.forEach((entry, field) => {
+            const isKey = field === '_key' || field === '_from' || field === '_to';
+            const category = entry.strings.size <= 10;
+            const number = entry.number === entry.total;
+            const date = entry.date === entry.total;
```

Comment:
What about missing values here? That would mean that a column with all numbers and one missing value would be set to 'label' below.

Reply:
In that case we'd have no choice but to recommend `label`--we can't coerce all the values to the target type.

Reply:
We can't convert to a type with some missing values for blank rows? For example, having a number column with 5 missing values. This is standard in most data manipulation libraries/tools

Reply:
This is a good point. So far we've assumed that data landing in Multinet is "clean" in this sense (i.e., no missing values, etc.). I am not sure if that assumption is code-enforced on the server (@AlmightyYakob probably knows the validation routines best). But whether it is or not, this deserves a deeper discussion. Would you mind filing an issue so we can figure this out the right way?

Reply:
I think most of the server-side validation is focused on tables matching each other (e.g. referenced keys existing), not so much column specifics. 

I think Jack's point is valid, one trivial way to address this would be to keep track of empty strings (kept track of in `TypeScore.empty`), and generate a type recommendation if that type count summed with the empty count equals the total. Example with number:

```
const number = entry.number + entry.empty === entry.total;
```

However this may be out of scope if we're just doing a first pass.

Reply:
I like this idea but yeah it will need to wait until we decide on the semantics/practice of handling missing values in our API. I've linked this discussion into https://github.com/multinet-app/multinet-rfcs/issues/16.